### PR TITLE
[Inference]Fix readme and example for API server

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
This PR fixed readme and examples for testing and using api server

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/server/README.md [hunk 1]
```diff
@@ -1,27 +1,33 @@
 # Online Service
-Colossal-Inference supports fast-api based online service. Simple completion and chat are both supported. Follow the commands below and
-you can simply construct a server with both completion and chat functionalities. For now we only support `Llama` model, we will fullfill
-the blank quickly.
+Colossal-Inference supports fast-api based online service. Simple completion and chat are both supported. Follow the commands below and you can simply construct a server with both completion and chat functionalities. For now we support `Llama`,`Glide_Llama` and `baichuan` model, we will fullfill the blank quickly.
```
Comment (yuanheng-zhao) [inline colossalai/inference/server/README.md:2]: When describing model names, there exist architectures (LlamaForCausalLM), model category (Llama, or more specifically, Llama, Llama2 and Llama 3), and model cards (meta-llama/Llama-2-7b). You could describes model categories here, such as "Llama 2, Llama 3, Baichuan 2, etc", and skip `Glide_Llama` as it's not a commonly used model structure.

File: colossalai/inference/server/api_server.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def add_engine_config(parser):
     # Parallel arguments not supported now
 
     # KV cache arguments
-    parser.add_argument("--block-size", type=int, default=16, choices=[8, 16, 32], help="token block size")
+    parser.add_argument("--block_size", type=int, default=16, choices=[8, 16, 32], help="token block size")
```
Comment (yuanheng-zhao) [inline colossalai/inference/server/api_server.py:136]: `8` is not supported.

File: colossalai/inference/server/api_server.py [hunk 3]
```diff
@@ -39,15 +38,16 @@
 app = FastAPI()
 
 
-# NOTE: (CjhHa1) models are still under development, need to be updated
-@app.get("/models")
-def get_available_models() -> Response:
-    return JSONResponse(supported_models_dict)
+@app.get("/ping")
```
Comment (yuanheng-zhao) [inline colossalai/inference/server/api_server.py:41]: Is it possible to add an engine check API to check if the engine is running properly?

File: colossalai/inference/server/README.md (hunk 1 above)
Comment (yuanheng-zhao) [inline colossalai/inference/server/README.md:2]: Please migrate this README to an appropriate section in colossalai/inference/README.md and remove this file.

Comment (CjhHa1) [reply colossalai/inference/server/README.md]: fixed

File: colossalai/inference/server/api_server.py (hunk 2 above)
Comment (CjhHa1) [reply colossalai/inference/server/api_server.py]: deleted

File: colossalai/inference/server/api_server.py (hunk 3 above)
Comment (CjhHa1) [reply colossalai/inference/server/api_server.py:41]: added

File: colossalai/inference/server/README.md (hunk 1 above)
Comment (CjhHa1) [reply colossalai/inference/server/README.md]: done

**Review by yuanheng-zhao — APPROVED** [review]

**Pull request merged**